frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/footer.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { useFlag } from '@console/shared/src/hooks/flag';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { ModalComponentProps } from '@console/internal/components/factory/modal';
+
+import { GUARDED_FEATURES } from '../../../features';
+import { POOL_PROGRESS } from '../../../constants/storage-pool-const';
+import {
+  BlockPoolAction,
+  BlockPoolActionType,
+  checkRequiredValues,
+} from '../../../utils/block-pool';
+
+export const BlockPoolModalFooter = (props: BlockPoolModalFooterProps) => {
+  const {
+    poolName,
+    poolStatus,
+    replicaSize,
+    volumeType,
+    dispatch,
+    onPoolCreation,
+    cancel,
+    close,
+  } = props;
+  const { t } = useTranslation();
+
+  const isPoolManagementSupported = useFlag(GUARDED_FEATURES.OCS_POOL_MANAGEMENT);
+
+  const handleTryAgainButton = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();
+    dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: '' });
+  };
+
+  const handleFinishButton = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();
+    if (poolStatus === POOL_PROGRESS.CREATED) {
+      onPoolCreation(poolName);
+    }
+    close();
+  };
+
+  const submitPoolCreation = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();
+    if (poolStatus === '') {
+      dispatch({ type: BlockPoolActionType.SET_IS_SUBMITTED, payload: true });
+    }
+  };
+
+  if (poolStatus) {
+    return (
+      <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
+        {poolStatus === POOL_PROGRESS.FAILED && (
+          <Button
+            type="button"
+            variant="secondary"
+            data-test="modal-try-again-action"
+            onClick={handleTryAgainButton}
+          >
+            {t('ceph-storage-plugin~Try Again')}
+          </Button>
+        )}
+        <Button
+          type="submit"
+          variant="primary"
+          data-test="confirm-action"
+          isDisabled={poolStatus === POOL_PROGRESS.PROGRESS}
+          id="confirm-action"
+          onClick={handleFinishButton}
+        >
+          {t('ceph-storage-plugin~Finish')}
+        </Button>
+      </ActionGroup>
+    );
+  }
+  return (
+    <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
+      <Button type="button" variant="secondary" data-test-id="modal-cancel-action" onClick={cancel}>
+        {t('ceph-storage-plugin~Cancel')}
+      </Button>
+      <Button
+        type="button"
+        variant="primary"
+        data-test="modal-create-action"
+        onClick={submitPoolCreation}
+        isDisabled={checkRequiredValues(
+          poolName,
+          replicaSize,
+          volumeType,
+          isPoolManagementSupported,
+        )}
+      >
+        {t('ceph-storage-plugin~Create')}
+      </Button>
+    </ActionGroup>
+  );
+};
```

Comment:
```suggestion
poolStatus ?
    (
      <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
        {poolStatus === POOL_PROGRESS.FAILED && (
          <Button
            type="button"
            variant="secondary"
            data-test="modal-try-again-action"
            onClick={handleTryAgainButton}
          >
            {t('ceph-storage-plugin~Try Again')}
          </Button>
        )}
        <Button
          type="submit"
          variant="primary"
          data-test="confirm-action"
          isDisabled={poolStatus === POOL_PROGRESS.PROGRESS}
          id="confirm-action"
          onClick={handleFinishButton}
        >
          {t('ceph-storage-plugin~Finish')}
        </Button>
      </ActionGroup>
    );
  : (
    <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
      <Button type="button" variant="secondary" data-test-id="modal-cancel-action" onClick={cancel}>
        {t('ceph-storage-plugin~Cancel')}
      </Button>
      <Button
        type="button"
        variant="primary"
        data-test="modal-create-action"
        onClick={submitPoolCreation}
        isDisabled={checkRequiredValues(
          poolName,
          replicaSize,
          volumeType,
          isPoolManagementSupported,
        )}
      >
        {t('ceph-storage-plugin~Create')}
      </Button>
    </ActionGroup>
  );
};
```

Reply:
ack

Reply:
ack

> Please update Readme of ceph package with added feature flag - 'pool-management'

can you please show me any example

Reply:
ack

> Please update Readme of ceph package with added feature flag - 'pool-management'

can you please show me any example